code/espurna/config/general.h

```diff
@@ -244,142 +198,135 @@ PROGMEM const char* const custom_reset_string[] = {
+#endif
 
 // Default pulse mode: 0 means no pulses, 1 means normally off, 2 normally on
+#ifndef RELAY_PULSE_MODE
 #define RELAY_PULSE_MODE            RELAY_PULSE_NONE
+#endif
 
 // Default pulse time in seconds
+#ifndef RELAY_PULSE_TIME
 #define RELAY_PULSE_TIME            1.0
+#endif
 
 // Relay requests flood protection window - in seconds
+#ifndef RELAY_FLOOD_WINDOW
 #define RELAY_FLOOD_WINDOW          3
+#endif
 
 // Allowed actual relay changes inside requests flood protection window
+#ifndef RELAY_FLOOD_CHANGES
 #define RELAY_FLOOD_CHANGES         5
+#endif
 
 // Pulse with in milliseconds for a latched relay
+#ifndef RELAY_LATCHING_PULSE
 #define RELAY_LATCHING_PULSE        10
+#endif
 
 // Do not save relay state after these many milliseconds
+#ifndef RELAY_SAVE_DELAY
 #define RELAY_SAVE_DELAY            1000
+#endif
 
-//------------------------------------------------------------------------------
-// LED
-//------------------------------------------------------------------------------
-
-#define LED_MODE_MQTT               0       // LED will be managed from MQTT (OFF by default)
-#define LED_MODE_WIFI               1       // LED will blink according to the WIFI status
-#define LED_MODE_FOLLOW             2       // LED will follow state of linked relay (check RELAY#_LED)
-#define LED_MODE_FOLLOW_INVERSE     3       // LED will follow the opposite state of linked relay (check RELAY#_LED)
-#define LED_MODE_FINDME             4       // LED will be ON if all relays are OFF
-#define LED_MODE_FINDME_WIFI        5       // A mixture between WIFI and FINDME
-#define LED_MODE_ON                 6       // LED always ON
-#define LED_MODE_OFF                7       // LED always OFF
-#define LED_MODE_RELAY              8       // If any relay is ON, LED will be ON, otherwise OFF
-#define LED_MODE_RELAY_WIFI         9       // A mixture between WIFI and RELAY, the reverse of MIXED
 
 // -----------------------------------------------------------------------------
 // WIFI
 // -----------------------------------------------------------------------------
 
+#ifndef WIFI_CONNECT_TIMEOUT
 #define WIFI_CONNECT_TIMEOUT    60000               // Connecting timeout for WIFI in ms
+#endif
+
+#ifndef WIFI_RECONNECT_INTERVAL
 #define WIFI_RECONNECT_INTERVAL 180000              // If could not connect to WIFI, retry after this time in ms
+#endif
+
 #define WIFI_MAX_NETWORKS       5                   // Max number of WIFI connection configurations
+
+#ifndef WIFI_AP_MODE
 #define WIFI_AP_MODE            AP_MODE_ALONE
+#endif
+
+#ifndef WIFI_SLEEP_MODE
 #define WIFI_SLEEP_MODE         WIFI_NONE_SLEEP     // WIFI_NONE_SLEEP, WIFI_LIGHT_SLEEP or WIFI_MODEM_SLEEP
+#endif
+
+#ifndef WIFI_SCAN_NETWORKS
 #define WIFI_SCAN_NETWORKS      1                   // Perform a network scan before connecting
+#endif
 
 // Optional hardcoded configuration (up to 2 networks)
 #ifndef WIFI1_SSID
 #define WIFI1_SSID              ""
-#endif
+
```

Comment:
`#endif`

Reply:
erm, right, copy & paste error. thanks for spotting it.